File: android/src/main/kotlin/com/example/t_stack/TStack.kt
package com.example.t_stack

import android.app.Activity
import android.content.Context
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.embedding.engine.FlutterEngineCache
import io.flutter.embedding.engine.dart.DartExecutor

object TStack {
    const val kEngineId = "t_stack_engine"

    internal var appContext: Context? = null
    lateinit var flutterEngine: FlutterEngine

    fun init(context: Context) {
        if (appContext == null) {
            appContext = context.applicationContext
            flutterEngine = FlutterEngine(context).also {
                it.dartExecutor.executeDartEntrypoint(DartExecutor.DartEntrypoint.createDefault())
                FlutterEngineCache.getInstance().put(kEngineId, it)
            }
        }
    }

    fun registerRoute(routeMap: Map<String, Class<out Activity>>) =
        TNavigator.registerRoute(routeMap)
}

File: ios/Classes/TStack.swift
let kEngineId = "t_stack_engine"

public class TStack {
    static public let shared = TStack()

    private init() {}

    public lazy var flutterEngine = FlutterEngine(name: kEngineId)

    var running = false

    public func initialize() {
        if (!running) {
            running = flutterEngine.run()
        }
    }

    public func registerRoute(_ routeMap: Dictionary<String, () -> UIViewController>) {
        TNavigator.shared.registerRoute(routeMap)
    }
}
